syntax = "proto3";

package microservice.template.v1;

option csharp_namespace = "Microservice.Template.V1";
option go_package = "github.com/example/microservice-template/gen/proto";
option java_multiple_files = true;
option java_package = "com.example.microservice.template.v1";
option objc_class_prefix = "MST";

service GreeterService {
  rpc SayHello(HelloRequest) returns (HelloResponse);
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
  rpc GenerateSampleData(SampleDataRequest) returns (SampleDataResponse);
  rpc AnalyzeCorrelation(CorrelationRequest) returns (AnalysisResponse);
  rpc AnalyzeClustering(ClusteringRequest) returns (AnalysisResponse);
  rpc AnalyzeDistribution(DistributionRequest) returns (AnalysisResponse);
}

message HelloRequest {
  string name = 1;
}

message HelloResponse {
  string message = 1;
  string trace_id = 2;
}

message HealthCheckRequest {}

message HealthCheckResponse {
  string status = 1;
  string version = 2;
  string uptime = 3;
}

// Analytics messages
message SampleDataRequest {
  int32 n_samples = 1;
  int32 seed = 2;
}

message SampleDataResponse {
  string data_csv = 1;
  string message = 2;
}

message CorrelationRequest {
  string data_csv = 1;
}

message ClusteringRequest {
  string data_csv = 1;
  int32 n_clusters = 2;
}

message DistributionRequest {
  repeated double values = 1;
  string column_name = 2;
}

message AnalysisResponse {
  bool success = 1;
  string message = 2;
  string results_json = 3;
  string plot_base64 = 4;
}
